wikidataintegrator/wdi_helpers/publication.py

```diff
@@ -192,20 +198,22 @@ def validate(self):
     def make_reference(self):
         if self.source == "crossref":
             assert 'doi' in self.ids
-            extid = wdi_core.WDString(self.ids['doi'], PROPS['DOI'], is_reference=True)
-            ref_url = wdi_core.WDUrl(self.ref_url, PROPS['reference URL'], is_reference=True)
-            stated_in = wdi_core.WDItemID(self.SOURCES[self.source], PROPS['stated in'], is_reference=True)
-            retrieved = wdi_core.WDTime(strftime("+%Y-%m-%dT00:00:00Z", gmtime()), PROPS['retrieved'],
-                                        is_reference=True)
-            self.reference = [stated_in, extid, ref_url, retrieved]
+            edt_id_id, ext_id_prop = self.ids['doi'], PROPS['DOI']
         elif self.source == "europepmc":
             assert 'pmcid' in self.ids
-            extid = wdi_core.WDString(self.ids['pmcid'], PROPS['PMCID'], is_reference=True)
-            ref_url = wdi_core.WDUrl(self.ref_url, PROPS['reference URL'], is_reference=True)
-            stated_in = wdi_core.WDItemID(self.SOURCES[self.source], PROPS['stated in'], is_reference=True)
-            retrieved = wdi_core.WDTime(strftime("+%Y-%m-%dT00:00:00Z", gmtime()), PROPS['retrieved'],
-                                        is_reference=True)
-            self.reference = [stated_in, extid, ref_url, retrieved]
```

Comment:
I noticed that these were exactly the same code duplicated besides two parts, so I refactored it before adding the `arxiv` changes